source/validate.cpp

```diff
@@ -332,14 +270,34 @@ void DebugInstructionPass(ValidationState_t& _,
   }
 }
 
+void ProcessIds(ValidationState_t& _, const spv_parsed_instruction_t& inst) {
+  if (inst.result_id) {
+    _.usedefs().AddDef(
+        {inst.result_id, inst.opcode,
+         std::vector<uint32_t>(inst.words, inst.words + inst.num_words)});
+  }
+  for (auto op = inst.operands; op != inst.operands + inst.num_operands; ++op) {
+    if (op->type == SPV_OPERAND_TYPE_ID)
+      _.usedefs().AddUse(inst.words[op->offset]);
+  }
+  if (inst.opcode == SpvOpEntryPoint) {
```

Comment:
This is not populating entry points correctly.   
The comment at the entry_points_ member says it's the list of IDs that are entry points.

So the right code here would be: 
```
if inst.opcode == SpvOpEntryPoint) { _.entry_points().push_back(inst.words[2]) }
```

The current code in the MR will add any IDs in the *interface* as entry points.

This is a case where the attempt to be general leads to confusing and incorrect code.  Just use the 2 as the index of the entry point ID.